### PR TITLE
Replace hard corded path sep with path.sep to support Windows

### PR DESCRIPTION
**Pull request by watilde**
Since the ci still have errors, let me fix the part of the causes -
https://github.com/npm/npm/pull/14045

### CODE REVIEW

File: entry.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ Entry.prototype.forContents = function (cb) {
 }
 
 Entry.prototype.computePath = function (entitypath) {
-  this.path = path.relative('/', entitypath)
+  this.path = path.relative(path.sep, entitypath)
```
Comment (iarna) [inline entry.js]: This should be fine on windows. (It turns out that `/` is treated identically to `\\` in Windows APIs.)

File: symlink.js [hunk 2]
```diff
@@ -10,14 +10,14 @@ module.exports = Symlink
 function Symlink (dest) {
   if (this == null) return new Symlink(dest)
   if (dest == null || dest === '') throw new Error('Symlinks must have a destination')
-  Entry.call(this, 'symlink', dest)
+  Entry.call(this, 'symlink', path.join(dest))
```
Comment (iarna) [inline symlink.js]: What is the `path.join` supposed to be doing here?

File: load-from-dir.js [hunk 3]
```diff
@@ -28,7 +28,7 @@ function loadFromDir (dir, top) {
     if (fileinfo.isSymbolicLink()) {
       var dest = fs.readlinkSync(filepath)
       if (/^(?:[a-z]\:)?[\\/]/i.test(dest)) {
-        dest = '/' + path.relative(top, dest)
+        dest = path.sep + path.relative(top, dest)
```
Comment (iarna) [inline load-from-dir.js]: This should be fine too.

Comment (iarna) [inline load-from-dir.js]: It's plausible that we should be normalizing the `path.relative` results however.

File: symlink.js [hunk 4]
```diff
@@ -10,14 +10,14 @@ module.exports = Symlink
...
 }
 inherits(Symlink, Entry)
 
 Symlink.prototype.create = function (where) {
   var filepath = path.resolve(where, this.path)
   var dest = this.contents
-  if (dest[0] === '/') {
+  if (dest[0] === path.sep) {
```
Comment (iarna) [inline symlink.js]: This could also be something like `C:\` which we'd want to treat as a match here.

File: test/functional.js [hunk 5]
```diff
@@ -79,7 +79,7 @@ test('loadFromDir', function (t) {
 
 test('generateFromDir', function (t) {
   var js = generateFromDir(testdir)
-  fs.writeFileSync(testmodule, js.replace(/'tacks'/g, "'../../index.js'"))
+  fs.writeFileSync(testmodule, js.replace(/'tacks'/g, "'.." + path.sep + ".." + path.sep + "index.js'"))
```
Comment (iarna) [inline test/functional.js]: `path.join` would be preferrable to `path.sep`, but `../../` should work fine on Windows.

File: symlink.js (hunk 2 above)
Comment (watilde) [inline symlink.js]: This `path.join` is to replace the path separator in the path with `path.sep`

File: load-from-dir.js (hunk 3 above)
Comment (watilde) [inline load-from-dir.js]: I've wrapped it with `path.join`

File: test/functional.js (hunk 5 above)
Comment (watilde) [inline test/functional.js]: true! I've updated.

**Pull request closed without merging**